cpp/src/arrow/flight/sql/odbc/flight_sql/CMakeLists.txt

```diff
@@ -0,0 +1,176 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+cmake_minimum_required(VERSION 3.16)
+set(CMAKE_CXX_STANDARD 17)
+
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+include_directories(include include/flight_sql
+                    ${CMAKE_SOURCE_DIR}/odbcabstraction/include)
+
+if(DEFINED CMAKE_TOOLCHAIN_FILE)
+  include(${CMAKE_TOOLCHAIN_FILE})
+endif()
+
+find_package(RapidJSON CONFIG REQUIRED)
+
+if(MSVC)
+  # the following definitions stop arrow from using __declspec when statically
+  # linking and will break on Windows without them
+  add_compile_definitions(ARROW_STATIC ARROW_FLIGHT_STATIC)
+endif()
+
+enable_testing()
+
+add_library(arrow_odbc_spi_impl
```

Comment:
Could you use our `add_arrow_lib()` instead?
https://github.com/apache/arrow/blob/598938711a8376cbfdceaf5c77ab0fd5057e6c02/cpp/cmake_modules/BuildUtils.cmake#L190

Reply:
We have an upcoming [PR#46099](https://github.com/apache/arrow/pull/46099/files#diff-ae8971934312f51437b00f3c4aa376bbf54aa0ce25a14abe1f74f3be7cd84486) that uses `add_arrow_lib(arrow_flight_sql_odbc)` to add an arrow library at the `odbc` directory. The `flight_sql` and `odbcabstraction` directories are subfolders to `arrow_flight_sql_odbc`, so I suggest we keep `add_library` for these 2 subfolders.

Reply:
Sorry, I may misunderstand your comment... What will be focused in #46099? Will #46099 focus on replacing `add_library()` with `add_arrow_lib()`?

> The `flight_sql` and `odbcabstraction` directories are subfolders to `arrow_flight_sql_odbc`, so I suggest we keep `add_library` for these 2 subfolders.

Could you explain why `add_library()` is better for them? (I'm not sure why "subfolders" is important here.)